openedx/core/djangoapps/user_api/views.py

```diff
@@ -178,6 +184,45 @@ def dispatch(self, request, *args, **kwargs):
         return super(RegistrationView, self).dispatch(request, *args, **kwargs)
 
 
+@transaction.non_atomic_requests
+@require_POST
+@outer_atomic(read_committed=True)
+def update_profile_info(request):
+    """
+    """
+    # Get the user
+
+    data = request.POST.copy()
+    user = request.user
+
+    post_registration_fields = [('gender', str), ('country', str), ('year_of_birth', int)]
```

Comment:
This check is good. But using a Django serializer with proper check is better.

Please add a GitHub issue to this repository so we can refactor it later.

Reply:
Done. https://github.com/Edraak/edraak-platform/issues/93